R/utils-get_code_dependency.R

```diff
@@ -287,28 +288,30 @@ extract_occurrence <- function(pd) {
     sym_cond <- setdiff(sym_cond, which(x$id %in% after_dollar))
   }
 
-  ass_cond <- grep("ASSIGN", x$token)
-  if (!length(ass_cond)) {
+  assign_cond <- grep("ASSIGN", x$token)
+  if (!length(assign_cond)) {
     return(c("<-", unique(x[sym_cond, "text"])))
   }
 
-  sym_cond <- sym_cond[sym_cond > ass_cond] # NOTE 1
+  # For cases like 'eval(expression(c <- b + 2))' removes 'eval(expression('.
+  sym_cond <- setdiff(
+    sym_cond,
+    sym_cond[sym_cond < min(assign_cond) & sym_cond %in% sym_fc_cond]
+  )
```

Comment:
Can the same be achieved with

```suggestion
  sym_cond <- sym_cond[!(sym_cond < min(assign_cond) & sym_cond %in% sym_fc_cond)]
```

Reply:
(optional)

This is a question of personal opinion on readability. 

Feel free to keep it the way you consider is best / consistent with `teal.code`!